static/app/views/settings/components/dataSecrecy/index.tsx

```diff
@@ -57,10 +59,33 @@ export default function DataSecrecy() {
         data: {allowSuperuserAccess: value},
       });
       setAllowAccess(value);
-      addSuccessMessage(t('Successfully updated access.'));
+
+      // if the user has allowed access, we need to remove the temporary access window
+      // only if there is an existing waiver
+      if (value && allowData) {
```

Comment:
problem 1 - if a user has a waiver set then allows access, we should delete the waiver. this is because after a while if the user turns out data secrecy again, we don't want them to get surprised with the old waiver.

hence we delete, so when the user enables data secrecy, they start fresh and need to create a new waiver.